evm/its.js

```diff
@@ -0,0 +1,723 @@
+const {
+    getDefaultProvider,
+    utils: { hexZeroPad },
+    Contract,
+} = ethers;
+const { Command, Option } = require('commander');
+const {
+    printInfo,
+    prompt,
+    printWarn,
+    printWalletInfo,
+    isValidAddress,
+    isKeccak256Hash,
+    wasEventEmitted,
+    mainProcessor,
+    isValidTokenId,
+    isValidNumber,
+    isString,
+    isValidCalldata,
+    isValidBytesAddress,
+    isNumberArray,
+} = require('./utils');
+const { getWallet } = require('./sign-utils');
+const IInterchainTokenService = require('@axelar-network/interchain-token-service/dist/interchain-token-service/InterchainTokenService.sol');
+const { addExtendedOptions } = require('./cli-utils');
+const tokenManagerImplementations = {
+    MINT_BURN: 0,
+    MINT_BURN_FROM: 1,
+    LOCK_UNLOCK: 2,
+    LOCK_UNLOCK_FEE: 3,
+};
+
+async function processCommand(chain, options) {
+    const { privateKey, address, action, yes } = options;
+
+    const contracts = chain.contracts;
+    const contractName = 'InterchainTokenService';
+    const contractConfig = contracts.InterchainTokenService;
+
+    const interchainTokenServiceAddress = address || contracts.interchainTokenService?.address;
+
+    if (!isValidAddress(interchainTokenServiceAddress)) {
+        throw new Error(`Contract ${contractName} is not deployed on ${chain.name}`);
+    }
+
+    const rpc = chain.rpc;
+    const provider = getDefaultProvider(rpc);
+
+    printInfo('Chain', chain.name);
+
+    const wallet = await getWallet(privateKey, provider, options);
+    const { address: walletAddress } = await printWalletInfo(wallet, options);
+
+    printInfo('Contract name', contractName);
+    printInfo('Contract address', interchainTokenServiceAddress);
+
+    const interchainTokenService = new Contract(interchainTokenServiceAddress, IInterchainTokenService.abi, wallet);
+
+    const gasOptions = contractConfig?.gasOptions || chain?.gasOptions || {};
+    printInfo('Gas options', JSON.stringify(gasOptions, null, 2));
+
+    printInfo('Action', action);
+
+    if (prompt(`Proceed with action ${action}`, yes)) {
+        return;
+    }
+
+    switch (action) {
+        case 'contractId': {
+            const contractId = await interchainTokenService.contractId();
+            printInfo('InterchainTokenService contract ID', contractId);
+
+            break;
+        }
+
+        case 'tokenManagerAddress': {
+            const tokenId = options.tokenId;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const tokenManagerAddress = await interchainTokenService.tokenManagerAddress(tokenIdBytes32);
+            printInfo(`TokenManager address for tokenId: ${tokenId}:`, tokenManagerAddress);
+
+            break;
+        }
+
+        case 'validTokenManagerAddress': {
```

Comment:
not needed, we can just print in the above method if token manager actually exists or not